ONSdigital/sdx-transform-cs
# Ensure sdx-transform-cs meets standards

**Changes**

- Add exceptions for failure to lookup survey schemas from disk
- Add exceptions for transformation failure when performing IO operations (ZIP/PDF/Buffer creation).
- Add failure and error logging for splunk/docker

**How to test**

- Send a message with invalid survey id within to any of the transformation endpoints. Check appropriate error is logged.
- Send invalid data to any of the transform endpoints. Check appropriate error is logged.
- Send a valid message payload and check that a success message is logged.

Testing io exceptions is more difficult. Prevent write access to the /tmp directory and run a transform.

**Who can test**

Anyone but @iwootten
